draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -140,9 +140,49 @@ This document uses "service" and "workload" interchangeably. Otherwise, all term
+
+[TODO: Should there be a DNS name to Trust domain mapping defined or some other discovery mechanism?]
+
+## Workload ID Token and Proof of Possession
+
+The Workload ID token (WIT) is bound to a secret cryptographic key and is always presented with a proof of possession as described in section [to-wit]. The WIT is a general purpose token that can be presented in multiple contexts. The WIT and its PoP are only used in the application-level options, and both are not used in MTLS. The WIT MUST NOT be used as a bearer token. While this helps reduce the sensitivity of the token it is still possible that a token and its proof of possession may be captured and replayed within the POP's lifetime. The following are some mitigations for the capture and reuse of the proof of possession (POP):
+
+* Preventing Eavesdropping and Interception with TLS
+
+An attacker observing or intercepting the communication channel can view the token and its proof of possession and attempt to replay it to gain an advantage. In order to prevent this the
+token and proof of possession MUST be sent over a secure, server authenticated TLS connection unless a secure channel is provided by some other mechanisms. Host name validation according
+to section [mutual-tls] MUST be performed. The WIT itself is not usable without a proof of possession.
+
+* Limiting Proof of Possession Lifespan
+
+The proof of possession MUST be time limited. A POP should only be valid over the time necessary for it to be successfully used for the purpose it is needed. This will typically be on the order of minutes.  POPs received outside their validity time MUST be rejected.
+
+* Limiting Proof of Possession Scope
+
+In order to reduce the risk of theft and replay the POP should have a limited scope. For example, a POP may be targeted for use with a specific workload and even a specific transaction to reduce the impact of a stolen POP. In some cases a workload may wish to reuse a POP for a period of time or have it accepted by multiple target workloads. A careful analysis is warranted to understand the impacts to the system if a POP is disclosed allowing it to be presented by an attacker along with a captured WIT.
+
+* Binding to a Timestamp or Nonce
+
+A proof of possession should include information that can be used to uniquely identify it such as a unique timestamp or nonce.  This can be used by the receiver to perform basic replay protection against tokens it has already seen. Depending upon the design of the system it may be difficult to synchronize the replay cache across all token validators. In this case, if the POP is not sufficiently scoped it may be usable with another workload.
+While a fresh nonce could be included in the POP, a mechanism for distributing a fresh challenge nonce from the validator to provide single use properties of a POP is outside the scope of this specification.
```

Comment:
```suggestion
The proof of possession MUST be time limited. A PoP should only be valid over the time necessary for it to be successfully used for the purpose it is needed. This will typically be on the order of minutes.  PoPs received outside their validity time MUST be rejected.

* Limiting Proof of Possession Scope

In order to reduce the risk of theft and replay the PoP should have a limited scope. For example, a PoP may be targeted for use with a specific workload and even a specific transaction to reduce the impact of a stolen PoP. In some cases a workload may wish to reuse a PoP for a period of time or have it accepted by multiple target workloads. A careful analysis is warranted to understand the impacts to the system if a PoP is disclosed allowing it to be presented by an attacker along with a captured WIT.

* Binding to a Timestamp or Nonce

A proof of possession should include information that can be used to uniquely identify it such as a unique timestamp or nonce.  This can be used by the receiver to perform basic replay protection against tokens it has already seen. Depending upon the design of the system it may be difficult to synchronize the replay cache across all token validators. In this case, if the PoP is not sufficiently scoped it may be usable with another workload.
While a fresh nonce could be included in the PoP, a mechanism for distributing a fresh challenge nonce from the validator to provide single use properties of a PoP is outside the scope of this specification.
```